### PR TITLE
feat(scan): add Astro monitoring component

### PR DESCRIPTION
**Pull request by xinyao27**
Add monitoring support for astro, usage:

```
// Layout.astro
---
import Monitoring from 'react-scan/monitoring/astro'
---

<Monitoring apiKey="" url="https://monitoring.react-scan.com/api/v1/ingest"/>
```

### CODE REVIEW

File: packages/scan/package.json [hunk 1]
```diff
@@ -200,9 +203,11 @@
     "auto.d.ts"
   ],
   "scripts": {
+    "copy-astro": "cp -R src/core/monitor/params/astro dist/core/monitor/params",
```
Comment (RobPruzan) [inline packages/scan/package.json:206]: why is this needed? Seems astro uses vite, so if there's nothing special that I'm not aware of, should be able to follow the same implementation as `react-scan/monitoring/react-router`

File: packages/scan/package.json [hunk 2]
```diff
@@ -51,6 +51,9 @@
       "import": "./dist/core/monitor/params/remix.mjs",
       "require": "./dist/core/monitor/params/remix.js"
     },
```
Comment (RobPruzan) [inline packages/scan/package.json:53]: Make sure to update the "typesVersions" too. It's needed for older versions of typescript that doesn't support the package.json exports field

**Review by RobPruzan — CHANGES_REQUESTED** [review]
Okay tested it and it works :) 

Last change I didn't already mention in the comments, can we make the Monitoring export from /monitor/astro a named export instead of default export just to stay consistent with the other framework exports

File: packages/scan/package.json (hunk 1 above)
Comment (xinyao27) [reply packages/scan/package.json:209]: packages/scan/src/core/monitor/params/astro/Monitoring.astro
packages/scan/src/core/monitor/params/astro/index.ts

The above two files will not be packaged, they will be exported directly and parsed by astro.
So these two files need to be manually copied into the dist directory.

**Comment by xinyao27** [comment]
> Okay tested it and it works :)
> 
> Last change I didn't already mention in the comments, can we make the Monitoring export from /monitor/astro a named export instead of default export just to stay consistent with the other framework exports

Updated, new way:

```
// Layout.astro
---
import { Monitoring } from 'react-scan/monitoring/astro'
---

<Monitoring apiKey="" url="https://monitoring.react-scan.com/api/v1/ingest"/>
```

**Review by RobPruzan — APPROVED** [review]
lgtm
